proposals/csharp-7.2/readonly-ref.md

```diff
@@ -164,125 +237,119 @@ static void ChangeV()
 
-However, since there is no backward compatibility considerations and there are workarounds, compiler should give a warning to ensure the implicit copying is noted by the user.
+The formal life time of temporary variables is semantically significant in scenarios involving escape analysis of variables returned by reference. 
 
 ## Metadata representation.
 
-We're using a combination of attributes and modifiers on signatures:
+When `System.Runtime.CompilerServices.IsReadOnlyAttribute` is applied to a byref parameter, it means that the the parameter is an `in` parameter.
 
-1) `IsReadOnlyAttribute` is added to the framework, to be used on parameters and return types, to indicate that they are `ref readonly`. For older frameworks, the compiler will generate an embedded (hidden) attribute with the same name into the module being built, and we will have a mechanism for disallowing using this attribute in source code.
-2) We will use a `modreq` with the framework type `IsConst`, to prevent other compilers and older versions of the C# compiler to read such signatures. This might not be needed on some cases like parameters of non-virtual methods, as downlevel use is safe. For tools and other compilers reading these metadata, they should always rely on the attribute, as it is mandatory.
+In addition, if the method is *abstract* or *virtual*, then the signature of such parameters (and only such parameters) must have `modreq[System.Runtime.CompilerServices.IsReadOnlyAttribute]`. 
 
-# Readonly ref returns
+**Motivation**: this is done to ensure that in a case of method overriding/implementing the `in` parameters match.
 
-"Readonly ref returns" is an augmentation of ref returns that allows returning references to variables without exposing them to modification.
+Same requirements apply to `Invoke` methods in delegates. 
 
-# Readonly ref locals
+**Motivation**: this is to ensure that existing compilers cannot simply ignore `readonly` when creating or assigning delegates.
 
-Locals are not supported for now. No plans to support them in this proposal.
+# Returning by readonly reference.
 
 ## Motivation
-The motivation for this sub-feature is roughly symmetrical to the reasons for the `in` parameters - avoiding copying, but on the returning side. A method or an indexer of a nontrivial struct type has currently two options -return by reference and be exposed to possible mutations or copy the value.
+The motivation for this sub-feature is roughly symmetrical to the reasons for the `in` parameters - avoiding copying, but on the returning side. A method or an indexer has currently two options: 1) return by reference and be exposed to possible mutations or 2) return by value which results in copying.
 
-## Solution
-`ref readonly` _returns_. (a more concise modifier like `in` so far has been elusive for this case).
+## Solution (`ref readonly` returns)
+(a more concise single modifier like `in` has been elusive for this case).
```

Comment:
Avoid parens here and just write a normal sentence.

Reply:
Not even sure if this is still relevant. I will probably just remove this. Or remove parens...

---
In reply to: [154466607](https://github.com/dotnet/csharplang/pull/1165#discussion_r154466607) [](ancestors = 154466607)